core/src/main/java/com/datastax/oss/driver/api/core/metadata/TokenMap.java

```diff
@@ -144,4 +144,8 @@
   default Set<Node> getReplicas(@NonNull String keyspaceName, @NonNull TokenRange range) {
     return getReplicas(CqlIdentifier.fromCql(keyspaceName), range);
   }
+
+  /** The name of the partitioner class in use, as reported by the Cassandra nodes. */
+  @NonNull
+  String getPartitionerName();
```

Comment:
Maybe instead of using `String`, we could introduce type like `PartitionerName` to make it more strong typed?

Reply:
I don't think we need strong type safety because this value is only exposed for reading, the user never passes it into the API.
Besides, the partitioner is pluggable(1), so there's no pre-defined set of values.

(1) at least in theory: you would have to write the server-side component and then plug-in your own `TokenFactoryRegistry` in the driver, that's a lot of work but feasible nevertheless.